docs/rfcs/001-project-specific-configuration.md

```diff
@@ -0,0 +1,126 @@
+
+```coffee
+# Global settings contents
+editor:
+  fontSize: 14
+```
+
+```coffee
+# Project settings contents
+editor:
+  fontSize: 16
+```
+
+```javascript
+let fontSize = atom.config.get('editor.fontSize')
+console.log(`fontSize = ${fontSize}`)
+// Outputs "fontSize = 16"
+```
+
+When setting or unsetting a configuration value, `atom.config.set()` and `atom.config.unset()` will continue to work the way it always has unless the `project` key is set to `true` in the options parameter. For example:
+
+```javascript
+// Sets the global editor.fontSize value to 18
+atom.config.set('editor.fontSize', 18)
+
+// Sets the project editor.fontSize value to 22
+atom.config.set('editor.fontSize', 22, {project: true})
+
+// Unsets the project core.fileEncoding value, leaving the global one unchanged
+atom.config.unset('core.fileEncoding', {project: true})
+```
+
+Setting a language-specific configuration can also be done at either the global or project setting level:
+
+```javascript
+// Sets the editor.tabLength setting to 4 for GitHub-flavored Markdown files globally
+atom.config.set('editor.tabLength', 4, {scopeSelector: '.source.gfm'})
+
+// Sets the editor.tabLength setting to 2 for GitHub-flavored Markdown files in this project
+atom.config.set('editor.tabLength', 2, {project: true, scopeSelector: '.source.gfm'})
+```
+
+**Note:** [Behavior is undefined](https://blogs.msdn.microsoft.com/oldnewthing/20140627-00/?p=633) when there are multiple project roots that have `atom-config.cson` files whose individual values are different. For example, given a project that has two project roots "foo" and "bar" with project settings like the following:
+
+```coffee
+# Project settings contents for project root "foo"
+editor:
+  fontSize: 16
+```
+
+```coffee
+# Project settings contents for project root "bar"
+editor:
+  fontSize: 22
+```
+
+```javascript
+// What will be returned is not defined
+let fontSize = atom.config.get('editor.fontSize')
+
+// Will return the default of "utf8"
+let fileEncoding = atom.config.get('core.fileEncoding')
+```
+
+Observing a configuration value will work similarly to reading a configuration value, largely unchanged except that change notifications may be raised when either the global or project setting is set, even if the merged value didn't change. For example:
+
+```coffee
+# Global settings contents
+editor:
+  fontSize: 14
+```
+
+```coffee
+# Project settings contents
+editor:
+  fontSize: 16
+```
+
+```javascript
+let fontSize = atom.config.get('editor.fontSize')
+console.log(`fontSize = ${fontSize}`)
+// "fontSize = 16" is output
+
+atom.config.observe('editor.fontSize', (value) => { console.log(`fontSize = ${value}`) })
+
+atom.config.set('editor.fontSize', 16)
+// "fontSize = 16" is output even though the value that would be returned at this point is still `16`
+```
+
+## Drawbacks
+
+It introduces complexity and some non-determinism into the Atom configuration system.
```

Comment:
Maybe explicitly note the potential for user confusion and increased support tickets?

Reply:
Added.